site/content/en/docs/Guides/feature-stages.md

```diff
@@ -0,0 +1,121 @@
+* Support for the overall feature will not be dropped, though details may change.
+* The schema and/or semantics of objects may change in incompatible ways in a subsequent beta or stable release. When
+  this happens, we will provide instructions for migrating to the next version. This may require deleting, editing,
+  and re-creating API objects. The editing process may require some thought. This may require downtime for
+  applications that rely on the feature.
+* Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases.
+  If you have multiple clusters that can be upgraded independently, you may be able to relax this restriction.
+
+> Note: Please do try `Beta` features and give feedback on them! After they exit beta, it may not be practical for us
+> to make more changes.
+
+### Stable
+
+A `Stable` feature means:
+
+* The feature is enabled and the corresponding feature gate no longer exists.
+* Stable versions of features will appear in released software for many subsequent versions.
+
+## Feature Stage Indicators
+
+There are a variety of features with Agones, how can we determine what stage each feature is in?
+
+Below are indicators for each type of functionality that can be used to determine the feature stage for a given aspect
+of Agones.
+
+### Custom Resource Definitions (CRDs)
+
+This refers to Kubernetes resource for Agones, such as `GameServer`, `Fleet` and `GameServerAllocation`.
+
+#### New CRDs
+
+For new resources, the stage of the resource will be indicated by the `apiVersion` of the resource.
+
+For example: `apiVersion: "agones.dev/v1"` is a `stable` resource, `apiVersion: "agones.dev/v1beta1"` is a `beta`
+ stage resource, and `apiVersion: "agones.dev/v1alpha1"` is a `alpha` stage resource.
+
+#### New CRD attributes
+
+`alpha` and `beta` attributes will have a corresponding `alpha` or `beta` parent element in their configuration to
+delineate their feature stage.
+
+If no such parent exists, this attribute is a `stable` feature.
+
+For example, if we were to add a hypothetical `alpha` configuration option of `timeoutSeconds` to the `GameServer`
+ resource, it would be configured like so:
+ 
+```yaml
+apiVersion: "agones.dev/v1"
+kind: GameServer
+metadata:
+  generateName: "simple-udp-"
+spec:
+  alpha: # this is the alpha feature block
+    timeoutSeconds: 30
+  ports:
+  - name: default
+    portPolicy: Dynamic
+    containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: simple-udp
+        image: gcr.io/agones-images/udp-server:0.15
+``` 
+
+As resource attributes progress through their stages, there may be breaking changes, as backward conversion between
+ `alpha`, `beta` and `stable` positioning in the resource configuration may not be guaranteed.
+
+### Agones Game Server SDK
+
+Any `alpha` or `beta` Game Server SDK functionality will have a comment block before each function declaring it either
```

Comment:
SDK path and its package should be reflective of its version. WDYT?

Reply:
Yeah - I'm not sure what the right answer here is. I honestly put this approach in here to hope someone would disagree with me :smile: 

Say we go by package -- maybe we do a subpackage? e.g. `sdk.alpha` ? I think it's trickier to do multiple top level packages across different languages -- but subpackages would solve the problems here with the current approach not being very obvious about which status of the SDK you are working against.

The only concern I can think of is that I don't think we can do multiple packages in our proto definition -- so while the user won't see this, we'll need to be clear in our comments on the proto definitions themselves.

How does that sound?

Reply:
> The only concern I can think of is that I don't think we can do multiple packages in our proto definition -- so while the user won't see this, we'll need to be clear in our comments on the proto definitions themselves.
> 
> How does that sound?

[This](https://cloud.google.com/apis/design/versioning) has some recommendations:

- "For all versions of the API other than v1, the major version number must be encoded as the last component of the proto package name. For example, google.pubsub.v2."

- "The minor and patch numbers should be reflected in the API configuration and documentation. They must not be encoded in the proto package name."

- "A stable API version must only depend on the latest stable version of other APIs."

Reply:
Waiting on resolution on #1043 before coming back to this section.